spinaltap-mysql/src/main/java/com/airbnb/spinaltap/mysql/MysqlSourceFactory.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2018 Airbnb. Licensed under Apache-2.0. See License in the project root for license
+ * information.
+ */
+package com.airbnb.spinaltap.mysql;
+
+import com.airbnb.spinaltap.common.source.Source;
+import com.airbnb.spinaltap.common.source.SourceState;
+import com.airbnb.spinaltap.common.util.Repository;
+import com.airbnb.spinaltap.common.validator.MutationOrderValidator;
+import com.airbnb.spinaltap.mysql.binlog_connector.BinaryLogConnectorSource;
+import com.airbnb.spinaltap.mysql.config.MysqlConfiguration;
+import com.airbnb.spinaltap.mysql.config.MysqlSchemaStoreConfiguration;
+import com.airbnb.spinaltap.mysql.schema.CachedMysqlSchemaStore;
+import com.airbnb.spinaltap.mysql.schema.LatestMysqlSchemaStore;
+import com.airbnb.spinaltap.mysql.schema.MysqlSchemaDatabase;
+import com.airbnb.spinaltap.mysql.schema.MysqlSchemaStore;
+import com.airbnb.spinaltap.mysql.schema.MysqlSchemaStoreManager;
+import com.airbnb.spinaltap.mysql.schema.MysqlSchemaTracker;
+import com.airbnb.spinaltap.mysql.schema.MysqlSchemaUtil;
+import com.airbnb.spinaltap.mysql.schema.MysqlTableSchema;
+import com.airbnb.spinaltap.mysql.schema.SchemaStore;
+import com.airbnb.spinaltap.mysql.schema.SchemaTracker;
+import com.airbnb.spinaltap.mysql.validator.EventOrderValidator;
+import com.airbnb.spinaltap.mysql.validator.MutationSchemaValidator;
+import com.github.shyiko.mysql.binlog.BinaryLogClient;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.validation.constraints.Min;
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+import org.skife.jdbi.v2.DBI;
+
+/** Represents a Factory for a {@link BinaryLogConnectorSource}. */
+@UtilityClass
+public class MysqlSourceFactory {
```

Comment:
This was previously a static factory method `MysqlSource.create`. Separated the logic for clarity and to improve readability. since it was substantial.